vizro-core/src/vizro/figures/kpi_cards.py

```diff
@@ -0,0 +1,156 @@
+"""Contains default KPI card functions."""
+
+import warnings
+from typing import Optional
+
+import dash_bootstrap_components as dbc
+import pandas as pd
+from dash import html
+
+from vizro.models.types import capture
+
+FORMATTING_WARNING = (
+    "Custom format string detected. Note that the format string is being evaluated, so ensure that "
+    "only trusted user input is provided to prevent potential security risks."
+)
+
+
+@capture("figure")
+def kpi_card(  # noqa: PLR0913
+    data_frame: pd.DataFrame,
+    value_column: str,
+    *,
+    value_format: str = "{value}",
+    agg_func: str = "sum",
+    title: Optional[str] = None,
+    icon: Optional[str] = None,
+) -> dbc.Card:
+    """Creates a styled KPI (Key Performance Indicator) card displaying a value.
+
+    Args:
+        data_frame: Data frame containing the data.
+        value_column: Column name of the value to be shown.
+        value_format: Format string to be applied to the value. It must be a valid Python format string where
+            `value` can be used. Defaults to "{value}".
+            For more details see: https://docs.python.org/3/library/string.html#format-specification-mini-language
+
+            Common examples include:
+             - "{value}": Displays the raw value.
+             - "${value:0.2f}": Formats the value as a currency with two decimal places.
+             - "{value:.0%}": Formats the value as a percentage without decimal places.
+             - "{value:,}": Formats the value with comma as a thousands separator.
+
+        agg_func: String function name to be used for aggregating the data. Common options include
```

Comment:
I was thinking about that before, but I thought it's better to simplify it here as the link contains all the possible options of configuring the `agg` function call, while we only support string function names. So to prevent any confusion I minimised it to string function names only and wanted to provide common examples. 

WDYT?

Reply:
Hm good question. Your typing indicates a string only, but I think in principle this would also work with any callable that fulfills this right?:

>Function to use for aggregating the data. If a function, must either work when passed a DataFrame or when passed to DataFrame.apply.

I don't mind too much, but I think what I don't like is that it says common options include xyz, but then it's unclear what would be allowed uncommon options. Do I guess function names?

Happy to go with whatever you choose. I would probably write just after: "For more information, please refer to ..."

Reply:
This is a great point @maxschulz-COL and I basically agree with both of you. It's actually pretty hard/impossible to find a definitive list of what strings work here in pandas - I've also wondered about this meany times before. It's nowhere in the official docs, and the system is basically yes, you just guess function names, and probably it will work 😆 The best reference I've found is this: https://stackoverflow.com/questions/65877567/passing-function-names-as-strings-to-pandas-groupby-aggregrate. So I would keep the text as is but add "For more information on possible functions, see https://stackoverflow.com/questions/65877567/passing-function-names-as-strings-to-pandas-groupby-aggregrate".

FYI it is deliberate that we only demonstrate how to do this with strings and keep the type-hint as `str` for exactly the reason @huong-li-nguyen says, so that things work in yaml straight away also. But I have deliberately chosen to not add in some validation that throws an exception if a callable is provided instead, because there's no reason to forbid that from working in Python and it would just make things more complicated than needed given these are meant to be simple copy and pastable functions.